### ISSUE TITLE
Are the builds fully reproducable?

### ISSUE THREAD

**Issue by commandodev** [nh2/static-haskell-nix#104]
I've just enabled CI with cachix in my new project:

https://github.com/cdodev/rapid/runs/1636776075

Even though I've done

`$(nix-build -A fullBuildScript --argstr projectDir $PWD)` followed by pushing to cachix locally, the CI run seems to be rebuilding the world.

@nh2 would you expect the dependencies to be pulled from cachix in this case?

I ran this command locally:

```
cachix push rapid /nix/store/mmqg7wiig0vyhy471hb8kmaplvdkvmab-rapid-0.1.0.0
compressing and pushing /nix/store/02a76sbyi6k112ha6nyvs02kpxka732b-libffi-3.3 (123.53 KiB)
compressing and pushing /nix/store/03id5md06a27kc92z0hz7b1a41z4bjdx-unix-compat-0.5.2 (136.80 KiB)
compressing and pushing /nix/store/173srxm01h19wy7ryfi1ks46wrik2h3s-quickcheck-io-0.2.0 (186.56 KiB)
compressing and pushing /nix/store/0qf8ca4326kicmq47zrkxbl51pyinsq8-string-conversions-0.4.0.1 (195.22 KiB)
compressing and pushing /nix/store/241bsz9mdxc0gwjz7mkm42ncwbxxkkk3-generics-sop-0.4.0.1 (20.15 MiB)
compressing and pushing /nix/store/0b4vvnnviqv1ckbvslcgqrjn09mfdyb1-swagger2-2.4 (24.30 MiB)
compressing and pushing /nix/store/0cgrashb7ivs33ndcfsyyc2mbdas4j1f-contravariant-1.5.2 (1.19 MiB)
compressing and pushing /nix/store/0fzahlddcdrg67jsqk24j23f991d8555-ghc-8.6.5-doc (252.51 MiB)
compressing and pushing /nix/store/1xpj60wr52jc1pilxz40qsma3ln84pmw-setenv-0.1.1.3 (27.98 KiB)
compressing and pushing /nix/store/08s90fjvdzfk0q5hm48ra5w8svlfc6wq-path-0.6.1 (688.13 KiB)
compressing and pushing /nix/store/1xyp61wyrqb1wadr9v0vqbhcbj7r1fzc-case-insensitive-1.2.0.11 (418.43 KiB)
compressing and pushing /nix/store/2r3ws4cx8ngxxc6kjyzddqikcmqf4ihy-uuid-types-1.0.3 (791.43 KiB)
compressing and pushing /nix/store/38x422ih7aybxkw457xnf5d9mg82a5m0-musl-1.1.24 (3.66 MiB)
compressing and pushing /nix/store/3g24xlnr0i2h1yi89dsg97ax9gdw4jwk-servant-server-0.16.2 (6.23 MiB)
compressing and pushing /nix/store/3k04wkp412cpjhiiklqcycmwf5f29kay-gcc-9.2.0 (129.77 MiB)
compressing and pushing /nix/store/3kz1gkp3i7wcsvgwigv64zqpcr2y1m7y-zlib-1.2.11 (133.07 KiB)
compressing and pushing /nix/store/3s1slyjh7jqbza6af05nvkrh6m00idm7-monad-control-1.0.2.3 (515.45 KiB)
compressing and pushing /nix/store/41ljadkf175n13f4mpdg8p1k78r8sfrh-singleton-bool-0.1.5 (142.92 KiB)
compressing and pushing /nix/store/45kdyh42imivfiv37pwzs2cb1h8cy00z-data-default-class-0.1.2.0 (178.09 KiB)
compressing and pushing /nix/store/46ra4qddcvf8m6g73ldqdn144ljgs1zy-mime-types-0.1.0.9 (7.20 MiB)
compressing and pushing /nix/store/4acf7dimh2s92jcrrn75i9ki8zhnl1sb-safe-exceptions-checked-0.1.0 (147.83 KiB)
compressing and pushing /nix/store/4m9rbriy52gdz629nskx53xc203a0f1q-easy-file-0.2.2 (377.15 KiB)
compressing and pushing /nix/store/4na1szan36krf06vd6gpnffz6zhwbw7q-dec-0.0.3 (112.12 KiB)
compressing and pushing /nix/store/4yci6sz4j33s0s5lq4cxh2pqn0mpc6x0-base64-bytestring-1.0.0.3 (354.38 KiB)
compressing and pushing /nix/store/50yyyrdc416drp104zn6kx97nglbb2pc-ncurses-6.1-20190112 (4.84 MiB)
compressing and pushing /nix/store/5m25m98ixxr8yhj870g9r0wflv7rn2hk-perl-5.30.1 (50.74 MiB)
compressing and pushing /nix/store/5nvqmfijxm5szplfgf62qq5jbv2ahc2p-linux-headers-4.19.16 (4.73 MiB)
compressing and pushing /nix/store/5xlcxmsmnscg3s88df8kgw6jvzarshzw-file-embed-lzma-0 (283.77 KiB)
compressing and pushing /nix/store/60qjxq482lf6xpj0zp3zgvz94sh7zrlj-transformers-base-0.4.5.2 (470.50 KiB)
compressing and pushing /nix/store/613j0vhk6x857w0x078cillq3xcnzvph-auto-update-0.1.6 (361.55 KiB)
compressing and pushing /nix/store/6a9xlcy9wa9kfw2fw1k116v6gjiskjff-cookie-0.4.5 (399.34 KiB)
compressing and pushing /nix/store/6r0gqz50kc3311dkd3fv4gwbmkwdlcxg-temporary-1.3 (169.97 KiB)
compressing and pushing /nix/store/6x3zlgha6a3xvwrmvd2masvzpx59y23m-tagged-0.8.6 (901.20 KiB)
compressing and pushing /nix/store/7484kkpngj4lr5kxm237ds8m2sxxrnv5-comonad-5.0.6 (1.65 MiB)
compressing and pushing /nix/store/79ip331hvkgghgfc4nyz6h048mibdsxc-primitive-0.6.4.0 (4.68 MiB)
compressing and pushing /nix/store/7b1s726qx7y62b93j0mbb9ykc7p1xb9q-th-lift-instances-0.1.14 (336.00 KiB)
compressing and pushing /nix/store/7cklg64jybqwkrzk6kzk2xbkpd8w3rlz-bash-4.4-p23 (1.24 MiB)
compressing and pushing /nix/store/7fligxkzp2ryl2ryr3hkpv8rzwll1rkv-servant-swagger-ui-core-0.3.3 (158.36 KiB)
compressing and pushing /nix/store/7mwh8fpzw4cmmdjdqpzijxbaqygj2ag6-gnugrep-3.3 (254.05 KiB)
compressing and pushing /nix/store/7nk0xigwslxhiagdhv7ymssbq9yw0cpz-mmorph-1.1.3 (719.01 KiB)
compressing and pushing /nix/store/7ppf6c43w26qwdxw217gqq3cazqh7j00-time-compat-1.9.2.2 (2.25 MiB)
compressing and pushing /nix/store/7v2bw1pwy5lbg44mdqx8sci2xlf8n87l-aeson-1.4.6.0 (25.82 MiB)
compressing and pushing /nix/store/828y6akmswhcq25dr41ryh7z6bil1aqv-scientific-0.3.6.2 (1.43 MiB)
compressing and pushing /nix/store/84kwq1mqd74nwiz1zgfv9fldwicrxwl9-warp-3.3.5 (4.02 MiB)
compressing and pushing /nix/store/89lwp1bnplcnml4ynciclj61m5nbzd0q-blaze-markup-0.8.2.3 (1.50 MiB)
compressing and pushing /nix/store/8by85zgjmvkjaf7l8vvj1p5imb85v3v9-gcc-9.2.0-lib (2.97 MiB)
compressing and pushing /nix/store/8davc1jj988yjzgrc4nlgslz0q6lwdna-musl-iconv-1.1.24 (528.00 B)
compressing and pushing /nix/store/8hv4n53x1k4aijz22xqi1p3q80gxf0fm-blaze-html-0.9.1.2 (22.71 MiB)
compressing and pushing /nix/store/8msbcalprb5d0bbi48y3w2xjcpan9xx9-file-embed-0.0.11.1 (353.02 KiB)
compressing and pushing /nix/store/8n933xy2zdpq2rm2j01sjmsidhd1w35j-kan-extensions-5.2 (3.15 MiB)
compressing and pushing /nix/store/8py8aldc5d4868d2d2i0z039vf255y7s-wai-3.2.2.1 (412.73 KiB)
compressing and pushing /nix/store/8zz8j2cbzg5x7kfy9f0x4apy1v0z1vbc-http-types-0.12.3 (1.95 MiB)
compressing and pushing /nix/store/90dk51dk5z41885wcmr5zw5ybhyz8x3s-fast-logger-2.4.17 (1.07 MiB)
compressing and pushing /nix/store/97rknlb54136pw7wg5p0wm5m93iypr3v-path-io-1.4.2 (924.54 KiB)
compressing and pushing /nix/store/97yy461w8vi70b43x8nybl4pvinl79a2-servant-blaze-0.9 (143.80 KiB)
compressing and pushing /nix/store/9ird077aviw00hq1pisff9lcqkjd2v88-lzma-0.0.0.3 (1.09 MiB)
compressing and pushing /nix/store/9kb6rik7vpp573av4msr2hpvarq78zs4-appar-0.1.8 (302.66 KiB)
compressing and pushing /nix/store/9va8vx68mhq8a6qajxxvi658pca6hc9v-parser-combinators-1.1.0 (659.75 KiB)
compressing and pushing /nix/store/9y8k5bdqmxb1kpa4wfqqcm84b9v11gdd-QuickCheck-2.13.2 (9.15 MiB)
compressing and pushing /nix/store/acnpwp2qrlckggh805hjl65pmqcf00kl-microlens-th-0.4.2.3 (977.02 KiB)
compressing and pushing /nix/store/avr2p8928n8cxfapa0c8smjwdx9k512x-iproute-1.7.8 (2.73 MiB)
compressing and pushing /nix/store/b7p7sv7hng9f3hiqwz6va405s4xrzyh0-tf-random-0.5 (881.44 KiB)
compressing and pushing /nix/store/b8k3wrpl6ydy0wdq5g28y3s98lidin0p-either-5.0.1.1 (471.19 KiB)
compressing and pushing /nix/store/bhaj1rd34f2vj32kknq2agmacr35ryhn-base-compat-0.10.5 (1.15 MiB)
compressing and pushing /nix/store/crbd2vdcb23as4n25sfv9fl89ks0r4sw-random-1.1 (1.57 MiB)
compressing and pushing /nix/store/czmii6721rwzbkfasz9cj342apihif84-base-compat-batteries-0.10.5 (1.22 MiB)
compressing and pushing /nix/store/d07wvzirxdr8s745qd843m2kr758czvn-streaming-commons-0.2.1.2 (2.89 MiB)
compressing and pushing /nix/store/d2w56gxd6isr4y8w2kn8jjvha8cssh57-hspec-2.7.1 (274.26 KiB)
compressing and pushing /nix/store/d5d6mycm222cwik7r72j6qx2cpj65hyw-void-0.7.3 (34.88 KiB)
compressing and pushing /nix/store/dg644iadmj3awsk6csyzcchkcmpb0msg-http-api-data-0.4.1.1 (3.62 MiB)
compressing and pushing /nix/store/dpcgh3q4i2qijgj4z4mld8px4pfxs75z-free-5.1.3 (10.60 MiB)
compressing and pushing /nix/store/f1bhhdcjk87i6mz2d34d0lsq2swk11la-musl-1.1.24-dev (545.49 KiB)
compressing and pushing /nix/store/f4755b01wwky0gk24r0vm4qsn2l6ljr8-aeson-pretty-0.8.8 (2.72 MiB)
compressing and pushing /nix/store/f5834nxs1wcmxhhsmas3pwjpdmqfzzd4-aws-lambda-haskell-runtime-2.0.3 (1.88 MiB)
compressing and pushing /nix/store/fawh84ljvwb0q3wykwla8vp5b9pf3ax3-pcre-8.43 (485.67 KiB)
compressing and pushing /nix/store/fr336a7h19c41wp099gcd373y0zlgdz4-megaparsec-7.0.5 (6.66 MiB)
compressing and pushing /nix/store/fydzgwa97q5mmin265dag9z9ijkd9qrd-lens-4.17.1 (31.63 MiB)
compressing and pushing /nix/store/g4gmg82g7297mbli9ysmr2qa1ra2dyqa-http2-2.0.3 (9.36 MiB)
compressing and pushing /nix/store/gg8jhq47861wlhkskb35fjg03rgr1pdk-parallel-3.2.2.0 (660.34 KiB)
compressing and pushing /nix/store/gglx9i4d8lrh5fp1m85xicas0hzw633q-string-conv-0.1.2 (372.04 KiB)
compressing and pushing /nix/store/gmdscsxz8dfagk80r1dncmrgvk24356w-gcc-wrapper-9.2.0 (40.48 KiB)
compressing and pushing /nix/store/gqcz9y1fjmrjjqbggi4ya74z4b8qk1n5-old-locale-1.0.0.7 (391.39 KiB)
compressing and pushing /nix/store/h77qmk2qfi4pf86h8b3h4hflr3lhibms-HUnit-1.6.0.0 (1.09 MiB)
compressing and pushing /nix/store/h9bfmy0x7g7842mp1407yqnynkcpbqns-sop-core-0.4.0.0 (3.49 MiB)
compressing and pushing /nix/store/hdh6z36jx3vn9kgpgnlp881s8n4b56cd-word8-0.1.3 (429.13 KiB)
compressing and pushing /nix/store/hyncmiab6r4f7vdy0qg2lv7ln39hznk3-colour-2.3.5-data (4.07 KiB)
compressing and pushing /nix/store/i2k0874600yhy26x56s80p2s9apflcxb-exceptions-0.10.4 (1.43 MiB)
compressing and pushing /nix/store/icbr5bnvrxh5l925p9m2gj84s3j3zzf7-unliftio-core-0.1.2.0 (239.38 KiB)
compressing and pushing /nix/store/iddpcfxsjknp3rjjn52b37i6kra5fhcz-hspec-core-2.7.1 (5.89 MiB)
compressing and pushing /nix/store/ifb4d5zs6p3hhb81ax8bid31vh88l7x0-attoparsec-iso8601-1.0.1.0 (319.44 KiB)
compressing and pushing /nix/store/ignfzlx8bzrrjbh5wz6mm9v406lm1alp-attr-2.4.48 (33.12 KiB)
compressing and pushing /nix/store/ipdq9ws7w54ajqk7xl2x7fsjc86ymk7c-unordered-containers-0.2.10.0 (3.03 MiB)
compressing and pushing /nix/store/ipl93ybbw7f664sqn0lcj8ql7xnk7zva-servant-swagger-ui-0.3.4.3.23.11 (1.35 MiB)
compressing and pushing /nix/store/j2jvz500wx60kvvdrvz3dbgrwarbc899-blaze-builder-0.4.1.0 (904.27 KiB)
compressing and pushing /nix/store/j3xsfd6qzydnlqsmg5v7qkdyy53g49iz-clock-0.8 (698.24 KiB)
compressing and pushing /nix/store/j84hwr2s4dllqq0m9b5p4x995554ww8c-semigroupoids-5.3.4 (6.20 MiB)
compressing and pushing /nix/store/j9zrk0pllc1396waqsm80mzpdkx82n5z-base-orphans-0.8.2 (785.28 KiB)
compressing and pushing /nix/store/jijifsm2gfwl20w9bx117gwsd4g6zbbl-libffi-3.3-dev (19.82 KiB)
compressing and pushing /nix/store/jn0pvrbyhbzlyz3bd5ncr8w4vdpp43fl-wai-app-static-3.1.7.1 (5.56 MiB)
compressing and pushing /nix/store/jwvg0q5sgngpx9krmmnzj9gmyhxiy6wp-http-media-0.8.0.0 (1.59 MiB)
compressing and pushing /nix/store/k1rchzfxrpfb5hbrsf0jwc738k8iqxqg-integer-logarithms-1.0.3 (312.95 KiB)
compressing and pushing /nix/store/kxwzc5zi062gb66rfjljvsr2qyp256il-ansi-wl-pprint-0.6.9 (829.06 KiB)
compressing and pushing /nix/store/l48vpnljh4nl9ds96y6d821jv0qkiwxr-cmdargs-0.10.20 (6.51 MiB)
compressing and pushing /nix/store/l5k67mj3cvpnggldriqdna8z3ay83pr8-unix-time-0.4.7 (466.08 KiB)
compressing and pushing /nix/store/ljaizcdcmgqjysbc0g35xf310db8qb0p-cryptonite-0.25 (27.49 MiB)
compressing and pushing /nix/store/lkbpxk8dz3p42lrfdyfdc20jlwvxiwwi-servant-0.16.2 (3.41 MiB)
compressing and pushing /nix/store/lnlva2cjcy65zqizlrgi1vphy8vcir3d-wai-extra-3.0.29 (3.55 MiB)
compressing and pushing /nix/store/ly06jfbk1glh7kkxc0qxl29q5whzpj7n-hashable-1.2.7.0 (1.20 MiB)
compressing and pushing /nix/store/m0d75cws0dbmx51zm0kgpkf687wb4vy5-invariant-0.5.3 (2.55 MiB)
compressing and pushing /nix/store/m2lkf6lj12f15xgh4hcm1235gjlqrn8d-hostname-1.0 (43.47 KiB)
compressing and pushing /nix/store/m8hg9k3byjvjjn80p8nfk7nl8ss1dp0g-transformers-compat-0.6.5 (1.02 MiB)
compressing and pushing /nix/store/m9j6az2dh4myhl1ydn02pwai3q0h6aqb-network-byte-order-0.1.4.0 (508.50 KiB)
compressing and pushing /nix/store/mincsgs9vl04y9swyybhcxv7dpc5hb7j-colour-2.3.5 (2.53 MiB)
compressing and pushing /nix/store/mk1i1k9zy0wjhykcw2qjmg3a9fy89arg-adjunctions-4.4 (3.26 MiB)
compressing and pushing /nix/store/mmqg7wiig0vyhy471hb8kmaplvdkvmab-rapid-0.1.0.0 (2.22 MiB)
compressing and pushing /nix/store/mpljv3za5i46mpz770gkq55br583ddna-servant-swagger-ui-redoc-0.3.3.1.22.3 (1.07 MiB)
compressing and pushing /nix/store/mwn9gbl4rwrdl23f9yk9ffg82nfiqnda-wai-logger-2.3.5 (290.19 KiB)
compressing and pushing /nix/store/ngj8jy8bjq5jgn5kdyxkhjmy9j3qhdp0-xz-5.2.4-bin (157.08 KiB)
compressing and pushing /nix/store/ni8gbgv0ym08fmqhzgvawy4spnf2gfjs-async-2.2.2 (787.30 KiB)
compressing and pushing /nix/store/nv3p86w6g2xk7a96qg9ba52n9nfhy28m-th-lift-0.8.1 (444.80 KiB)
compressing and pushing /nix/store/nz55rw3iq83aipkiznm7s13wgwg6nldm-th-abstraction-0.3.2.0 (2.46 MiB)
compressing and pushing /nix/store/p1ax1w0pnir0wgc29ql8kq1nismaivk7-bsb-http-chunked-0.0.0.4 (71.99 KiB)
compressing and pushing /nix/store/p7fdh2mzlvpkijpsz7vrsbd93dh533w2-insert-ordered-containers-0.2.2 (2.49 MiB)
compressing and pushing /nix/store/pbjm57wkqvncfy75xdcflkxjxndck5qk-psqueues-0.2.7.2 (2.25 MiB)
compressing and pushing /nix/store/pmail2mg8ybrygr1m3izzc0klvr1mgkz-gmp-6.1.2 (2.16 MiB)
compressing and pushing /nix/store/pna3ml3fn9z7p1rsl8nnqi3l9w6hwghz-time-manager-0.0.0 (172.40 KiB)
compressing and pushing /nix/store/pp6mwilzj84whjg6y53k3xhhar416ani-optparse-applicative-0.14.3.0 (3.87 MiB)
compressing and pushing /nix/store/pq9mhyaq7zskxwnnk2lim1sbnbxm7dbs-basement-0.0.11 (15.70 MiB)
compressing and pushing /nix/store/pxix17sqwfffw13pflr49wxy2y22g1v9-http-date-0.0.8 (555.66 KiB)
compressing and pushing /nix/store/pxiybc170y0mx7783frr9pb5p141ippj-xz-5.2.4-dev (170.66 KiB)
compressing and pushing /nix/store/q300mrz92aphx9w6rgvcrqyymz4b904c-hspec-discover-2.7.1 (1.86 MiB)
compressing and pushing /nix/store/qfbihq0n0mb6284h7l0q7nfr23ky5g9y-coreutils-8.31 (1.63 MiB)
compressing and pushing /nix/store/qh012q3yxi98za4vbvcwbwarjf2b8sks-memory-0.14.18 (3.49 MiB)
compressing and pushing /nix/store/qy56pj9kw2q0s397jl3w1csf639jqah9-binutils-2.31.1 (32.40 MiB)
compressing and pushing /nix/store/r2165z1allxi1lqzqrzhx9c23zcvzjyi-vector-0.12.1.2 (21.09 MiB)
compressing and pushing /nix/store/r421fyzmdi3j5418kwvgyb0lh5daz1mz-safe-exceptions-0.1.7.0 (435.88 KiB)
compressing and pushing /nix/store/r62ly3lx3cngqzqs9nj1brar3cyxmgij-attoparsec-0.13.2.3 (7.29 MiB)
compressing and pushing /nix/store/ricid9igi8nyxhz671zafangd5c8wckn-vault-0.3.1.3 (553.19 KiB)
compressing and pushing /nix/store/rjmbaz9yvxb0i9imvxk7x3nms90khzip-semigroups-0.18.5 (95.06 KiB)
compressing and pushing /nix/store/s1dh62a668x61lcswlsx9ymwg9qk20if-reflection-2.1.5 (4.06 MiB)
compressing and pushing /nix/store/sb4ykhbyfavjgskb995pf69lmj8xhvsr-profunctors-5.3 (5.47 MiB)
compressing and pushing /nix/store/sbm6mza33hcd6n0sjg9hkbapyy1gqbr7-zlib-0.6.2.1 (2.21 MiB)
compressing and pushing /nix/store/sg0vy7hbg2v9q8iwd424xh31pmy5lvp7-microlens-0.4.10 (814.73 KiB)
compressing and pushing /nix/store/swyawi8hr83qfg334y4z8vp66jgjpda0-servant-swagger-1.1.7.1 (1.19 MiB)
compressing and pushing /nix/store/v905dxdbxmzi23sa4l9pa9p9qcyqf4lr-mtl-2.2.2 (1.55 MiB)
compressing and pushing /nix/store/v9m63d5v66ldsq75cvxsx0iqv9nypv47-call-stack-0.1.0 (164.27 KiB)
compressing and pushing /nix/store/vb2kkspz6lpdjfp7g56640s410lsfw96-network-uri-2.6.2.0 (1.85 MiB)
compressing and pushing /nix/store/vb8y1jlizzd7fzb47hmn9scinx5f76ps-utf8-string-1.0.1.1 (950.40 KiB)
compressing and pushing /nix/store/vfh4hrwjc9k7b6n2zizc03s1xack4pyk-dlist-0.8.0.7 (352.09 KiB)
compressing and pushing /nix/store/vn39a07hrpvljgfp80skix0n0xd1c1gr-old-time-1.1.0.3 (1.36 MiB)
compressing and pushing /nix/store/vni3f0d60vjmcbm4zq5ifr9y4rli3ijy-binutils-wrapper-2.31.1 (37.14 KiB)
compressing and pushing /nix/store/vrnh0xxic450r4bs0vjcraim7hcbhybn-gmp-6.1.2-dev (196.63 KiB)
compressing and pushing /nix/store/vyf9l454ip0a9km0nqd0mcj8ijb6vhha-network-2.8.0.1 (4.38 MiB)
compressing and pushing /nix/store/vyvqxg0kl21px6mr8a1lm960p4hgclbc-unliftio-0.2.12 (3.45 MiB)
compressing and pushing /nix/store/qpk6kf6z7n9y9yxk4s2d5zh4nr19v0wx-hspec-expectations-0.8.2 (428.38 KiB)
compressing and pushing /nix/store/vzqs9p321m39dm4ab1rm2yl90dd7nkb8-byteorder-1.0.4 (161.18 KiB)
compressing and pushing /nix/store/w75fwx4a96zqbw1x12g33ib2j9g5hj97-splitmix-0.0.3 (372.44 KiB)
compressing and pushing /nix/store/wxwq0g6fw237z1xzj25lkykmlfygxihi-zlib-1.2.11 (258.41 KiB)
compressing and pushing /nix/store/x1kcjwlc5ykhns1qvkcpj5464m1vx4cf-ncurses-6.1-20190112 (3.38 MiB)
compressing and pushing /nix/store/xd86b1352xlndqhlsbnm9xww41j0cy9q-neat-interpolation-0.3.2.6 (759.88 KiB)
compressing and pushing /nix/store/xhzljp348wgsvmiyjnlvf8s4nnz7dc37-acl-2.2.53 (45.65 KiB)
compressing and pushing /nix/store/xjnn4h0z6d7ycwzwr663c86fm2qjy3m4-ansi-terminal-0.9.1 (2.19 MiB)
compressing and pushing /nix/store/xkalz816wbn74pggrzcymipryvxdi5ja-zlib-1.2.11-dev (111.77 KiB)
compressing and pushing /nix/store/xrm7264642881ia9iznp99wffxsw7dh5-parsec-3.1.14.0 (3.65 MiB)
compressing and pushing /nix/store/xrp57ywvgvnqwsg8plqgvz8c7pc4wbv3-ghc-8.6.5 (1.37 GiB)
compressing and pushing /nix/store/y54w6zalrqnjvx58ifs6ka3crs7gjl6i-bifunctors-5.5.7 (7.41 MiB)
compressing and pushing /nix/store/y9ky4vca6r8akaxv4gy5p5iadnnhlsnk-resourcet-1.2.2 (1.27 MiB)
compressing and pushing /nix/store/yfg5v8w48n71px93ijn51lr2cam1n1az-expand-response-params (16.33 KiB)
compressing and pushing /nix/store/yfy58bkvz58xybc4zr1dkx8m91gfcr4y-katip-0.8.3.0 (4.64 MiB)
compressing and pushing /nix/store/z1qaajy94a95jdixbgzipc9chssqn223-distributive-0.6.1 (686.50 KiB)
compressing and pushing /nix/store/z579cmk2i3lxly9175wykzavs2j7hd27-xz-5.2.4 (468.57 KiB)
compressing and pushing /nix/store/z691varvkd0910j5s2sfdfcni06kblzh-http-client-0.6.4 (5.33 MiB)
compressing and pushing /nix/store/z9hbky3w454xrlaim6rxh7qnb0h2qic2-simple-sendfile-0.2.30 (206.14 KiB)
compressing and pushing /nix/store/zdrg32j03zjg9v2qjxpp0cf2ay9j9fx5-StateVar-1.2 (254.52 KiB)
compressing and pushing /nix/store/zyiadx2h2wakzv13zry70jdznxb2l9sr-lens-aeson-1.0.2 (1.87 MiB)
All done.
```

**Comment by nh2** [comment]
I believe one problem is that when cachix went out of beta, the Free for Open Source cachixes were trimmed to 10 GB. The static-haskell-nix Cachix before had way more than 100 GB.

The my static-haskell-nix CI server has automatic cachix pushing enabled, but it builds against nixpkgs master every night, so it will create a large amount of builds. The `/nix/store` on that is 1.5 TB currently, and all of that is pushed to cachix -- but only the most recent 10 GB are cached there, which explains bad cache hit rates.

For example, `/nix/store/xrp57ywvgvnqwsg8plqgvz8c7pc4wbv3-ghc-8.6.5` exists on the CI server, so it was certainly uploaded to cachix in the past, but now is no longer there. I've manually re-uploaded it now.

In the future, it may make sense for me to offer the CI server directly as a binary cache, so that users can make use of the full 1.5 TB of builds, and not only 10 GB.

Finally, if you're using `stack2nix`, then of course each `resolver` may result in different packages. So it is expected that the GHCs are cached (which also take the longest time), but the individual packages not.

In summary, it's a cache size issue, not a reproducibility problem.

**Comment by commandodev** [comment]
Thanks @nh2!

I've run `cachix push` locally, so I had assumed that I would have pushed every dependency of `rapid` to the cache. @domenkozar is that the right mental model here? I assume I must have done something wrong as the rapid cache only has 266.3 MB in it...

**Comment by commandodev** [comment]
A little more investigating:

I've cleared the rapid cache and I've unchecked the upstream option. I've run

`$(nix-build -A fullBuildScript --argstr projectDir $PWD)`

And then

`cachix push -j10 rapid /nix/store/mmqg7wiig0vyhy471hb8kmaplvdkvmab-rapid-0.1.0.0` (which is the resulting store path)

I'm now building on a separate machine (my nixos laptop), with `cachix use {rapid,static-haskell-nix}`. Initial impressions are that it's building the world again.

Will report back when it's finished!

**Comment by nh2** [comment]
@commandodev Just in case: To check nix-level input reproducibility, you don't need to finish the build, just compare the rapid `.drv` file paths that `nix-build` prints it will build (and if different, `nix-diff` them).

Depending of how you specify `src`, it is easy for e.g. the `.git` directory or similar nonreproducible files to make it in, and `nix-diff` will help point out which those are.

**Comment by commandodev** [comment]
I've not idea what was different between my machine and the run on github, but letting it run through and push from there seems to have worked:

https://github.com/cdodev/rapid/runs/1640209767 `succeeded 1 minute ago in 4m 51s`

I'd like to get this down a bit if possible, so going to look at caching `/nix` (and perhaps the stack directory) itself to see if I can get any speed up
